src-tauri/src/filesystem/mod.rs

```diff
@@ -11,6 +14,14 @@ pub const fn bytes_to_gb(bytes: u64) -> u16 {
     (bytes / (1e+9 as u64)) as u16
 }
 
+/// Opens a file at the given path. Returns true if was successful
+#[tauri::command]
+pub async fn open_file(path: String) -> Result<bool, ()> {
+    Ok(tokio::task::spawn_blocking(move || {
+        open::commands(path)[0].status().is_ok()
+    }).await.unwrap_or(false)) // NOTE(conaticus): I tried handling the errors nicely here but Tauri was mega cringe and wouldn't let me nest results in async functiona
+}
```

Comment:
Note about this - I wasn't sure how to handle this and tried a few different methods to return a more informative result however Tauri restricted me on nesting Results, which was required in an async function.